domain/model/state/types.go

```diff
@@ -430,3 +432,96 @@ type dbModelState struct {
 	CredentialInvalidReason string `db:"cloud_credential_invalid_reason"`
 	Migrating               bool   `db:"migrating"`
 }
+
+type dbModelConstraint struct {
+	ModelUUID      string `db:"model_uuid"`
+	ConstraintUUID string `db:"constraint_uuid"`
+}
+
+type dbConstraint struct {
+	UUID             string  `db:"uuid"`
+	Arch             *string `db:"arch"`
+	CPUCores         *uint64 `db:"cpu_cores"`
+	CPUPower         *uint64 `db:"cpu_power"`
+	Mem              *uint64 `db:"mem"`
+	RootDisk         *uint64 `db:"root_disk"`
+	RootDiskSource   *string `db:"root_disk_source"`
+	InstanceRole     *string `db:"instance_role"`
+	InstanceType     *string `db:"instance_type"`
+	ContainerType    *string `db:"container_type"`
+	VirtType         *string `db:"virt_type"`
+	AllocatePublicIP *bool   `db:"allocate_public_ip"`
+	ImageID          *string `db:"image_id"`
+}
+
+func (c dbConstraint) toValue(tags []dbConstraintTag, spaces []dbConstraintSpace, zones []dbConstraintZone) (constraints.Value, error) {
+	if c.UUID == "" {
+		return constraints.Value{}, nil
+	}
+
+	consVal := constraints.Value{
+		Arch:           c.Arch,
+		CpuCores:       c.CPUCores,
+		CpuPower:       c.CPUPower,
+		Mem:            c.Mem,
+		RootDisk:       c.RootDisk,
+		RootDiskSource: c.RootDiskSource,
+		InstanceRole:   c.InstanceRole,
+		InstanceType:   c.InstanceType,
+		VirtType:       c.VirtType,
+		ImageID:        c.ImageID,
+	}
+	if c.AllocatePublicIP != nil && *c.AllocatePublicIP {
```

Comment:
This is not code or logic that we have today and potentially changes the meaning of the value. Do you have a reference as to the why we are doing this?